mbuild/box.py

```diff
@@ -2,151 +2,343 @@
+        Vectors that define a right-handed parallelepiped (triclinic box).
+    precision : int, optional, default=None
+        Control the precision of the floating point representation __repr__
+
     Attributes
     ----------
-    mins : np.ndarray, shape=(3,), dtype=float
-        Minimum x, y, z coordinates.
-    maxs : np.ndarray, shape=(3,), dtype=float
-        Maximum x, y, z coordinates.
-    lengths : np.ndarray, shape(3,), dtype=float
-        Box length in x, y and z directions.
-    angles : np.ndarray, shape(3,), dtype=float, default=[90,90,90]
-        Angles defining the tilt of the box
+    box_vectors : np.ndarray, shape=(3,3), dtype=float
+        Vectors that define the parallelepiped (Box).
+    Lx : float
+        Length of the Box in the x dimension
+    Ly : float
+        Length of the Box in the y dimension
+    Lz : float
+        Length of the Box in the z dimension
+    xy : float
+        Tilt factor needed to displace an orthogonal box's xy face to its parallelepiped structure.
+    xz : float
+        Tilt factor needed to displace an orthogonal box's xz face to its parallelepiped structure.
+    yz : float
+        Tilt factor needed to displace an orthogonal box's yz face to its parallelepiped structure.
+    precision : int
+        Precision of the floating point numbers when accessing the __repr__
 
+    NOTE
+    ----
+    Box vectors are expected to be provided in row-major format.
     """
-    def __init__(self, lengths=None, mins=None, maxs=None, angles=None):
-        if lengths is None:
-            if mins is None or maxs is None:
-                raise ValueError(
-                    "Either provide `lengths` or `mins` and `maxs`. "
-                    "You provided: "
-                    "lengths={} mins={} maxs={}".format(lengths, mins, maxs)
-                )
-            mins = np.array(mins, dtype=float)
-            maxs = np.array(maxs, dtype=float)
-            assert mins.shape == (3, ), "Given mins have wrong dimensions"
-            assert maxs.shape == (3, ), "Given maxs have wrong dimensions"
-            assert all(mins <= maxs), "Given mins are greater than maxs"
-            self._mins = _BoxArray(array=mins, var="mins", box=self)
-            self._maxs = _BoxArray(array=maxs, var="maxs", box=self)
-            self._lengths = _BoxArray(array=(self.maxs - self.mins), var="lengths", box=self)
+
+    def __init__(self, box_vectors=None, precision=None):
+        box_vectors = _validate_box_vectors(box_vectors)
+
+        self._box_vectors = box_vectors
+
+        (Lx, Ly, Lz, xy, xz, yz) = self._from_vecs_to_lengths_tilt_factors()
+        self._Lx = Lx
+        self._Ly = Ly
+        self._Lz = Lz
+        self._xy = xy
+        self._xz = xz
+        self._yz = yz
+
+        if precision is not None:
+            self._precision = int(precision)
+        else:
+            self._precision = precision
+
+    @classmethod
+    def from_lengths_angles(cls, lengths, angles):
```

Comment:
```suggestion
    def from_lengths_angles(cls, lengths, angles=[90, 90, 90]):
```
It might be nice if the angles had a default set to 90 to save some typing for users. (I know usually I'm using cuboid boxes.)